server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -718,10 +718,49 @@ public Map<String, UploadedSegmentMetadata> getSegmentsUploadedToRemoteStore() {
         return Collections.unmodifiableMap(this.segmentsUploadedToRemoteStore);
     }
 
+    // Visible for testing
+    Set<String> getMetadataFilesToFilterActiveSegments(
```

Comment:
I am finding this method very difficult to understand. Is there a way we can simplify this?

Reply:
this is the simplified version :), 
the idea here is for each deletable md file, we can consider the segments present in non-deletable md file before this and non-deletable md file after this as the segments which cannot be deleted.